### PR TITLE
Update dependencies and switch to haskell.nix cabalProject

### PR DESCRIPTION
**Pull request by angerman**

### CODE REVIEW

**Review by manveru — APPROVED** [review]

File: src/cddl-test/Main.hs [hunk 1]
```diff
@@ -107,6 +110,11 @@ dbArgs fp = do
     , immCheckIntegrity = const True -- No validation
     , immHasFS          = ioHasFS $ MountPoint (fp </> "immutable")
     , immTracer         = nullTracer
+    , immCacheConfig    = CacheConfig
+        { pastEpochsToCache = 1
+        , expireUnusedAfter = 42
```
Comment (erikd) [inline src/cddl-test/Main.hs:115]: Assuming these magic values are ok for a test. Worst thing than can happen is a test fails.

File: src/cddl-test/Main.hs [hunk 2]
```diff
@@ -107,14 +110,20 @@ dbArgs fp = do
...
     }
 
 data ReadResult = Block ByteString | FutureEpoch | NoData
   deriving (Show, Eq)
 
 readBlock :: ImmDB IO ByronBlock -> EpochNo -> IO ReadResult
 readBlock db epoch = do
-    ret <- try $ getBlob db ChainDB.Block $ Left epoch
+    ret <- try $ ChainDB.getBlockComponent db ChainDB.GetRawBlock $ Left epoch
+    --ret <- try $ getBlob db ChainDB.Block $ Left epoch
```
Comment (erikd) [inline src/cddl-test/Main.hs:126]: We should remove that comment.

**Review by erikd — APPROVED** [review]
We should probably remove that commented out line of code, but otherwise LGTM!

File: src/cddl-test/Main.hs [hunk 3]
```diff
@@ -24,15 +24,17 @@ import           Control.Tracer (nullTracer)
 import qualified Cardano.Crypto as Cardano (ProtocolMagicId)
```
Comment (avieth) [inline src/cddl-test/Main.hs:24]: Somebody made this thing build? Nice. I had just set it to `buildable: False`.

**Review by avieth — APPROVED** [review]
LGTM.

The cddl-test thing is not important, and apparently it uses interfaces from ouroboros-consensus that are not intended to be used.

Is the nix end-user interface significantly different? I heard it's no longer `-A nix-tools.{exes|libs|...}` but instead `-A haskellPackages.cardano-byron-proxy.components.{exes|libs|...}`. Any other big changes? Is it now easy to get a proper nix-shell?

**Pull request merged**